pytorch_lightning/trainer/callback_hook.py

```diff
@@ -197,14 +199,29 @@ def on_keyboard_interrupt(self):
         for callback in self.callbacks:
             callback.on_keyboard_interrupt(self, self.lightning_module)
 
-    def on_save_checkpoint(self):
+    @staticmethod
+    def __is_old_signature(fn: Callable) -> bool:
```

Comment:
can we make this a normal function that can be used anywhere, just in-case we need to deprecate more signatures? it can take old parameters count (`2`) as input.

Reply:
I don't think there is a point right now, as we don't know how useful it will be for other deprecations.

If a future PR could use it, the author can take this and move it outside if necessary.